src/main/java/org/fenixedu/ulisboa/applications/services/cms/site/alternativesite/AlternativeSitesImporter.java

```diff
@@ -77,83 +73,65 @@ public ProcessResult processFile(File file, ExecutionInterval executionInterval)
+                } else {
+                    this.result.addErrorMessage(
+                            "label.org.fenixedu.ulisboa.applications.management.cms.site.alternativesite.process.error.invalidLine",
+                            String.valueOf(row.getRowNum() + 1));
                 }
-            }
+            });
         }
     }
-
-    private void processSheet(Sheet sheet, ExecutionInterval executionInterval, ProcessResult result) {
-        Iterator<Row> rowIterator = sheet.rowIterator();
-
-        // Skip header row
-        rowIterator.next();
-
-        while (rowIterator.hasNext()) {
-            final Row row = rowIterator.next();
-            processRow(row, executionInterval, result);
-        }
-    }
-
-    private void processRow(Row row, ExecutionInterval executionInterval, ProcessResult result) {
-        if (row.getPhysicalNumberOfCells() >= NUMBER_REQUIRED_COLUMS) {
-            Degree degree = getDegree(row, result);
-            if (degree == null) {
-                return;
-            }
-
-            DegreeCurricularPlan degreeCurricularPlan = getDegreeCurricularPlan(row, degree, result);
-            if (degreeCurricularPlan == null) {
-                return;
-            }
-
-            ExecutionCourse executionCourse = getExecutionCourse(row, degreeCurricularPlan, executionInterval, result);
-            if (executionCourse == null) {
-                return;
-            }
-
-            if (executionCourse.getSite() == null) {
-                result.addErrorMessage(
-                        "label.org.fenixedu.ulisboa.applications.management.cms.site.alternativesite.process.error.noSiteConfigured",
-                        String.valueOf(row.getRowNum() + 1), executionInterval.getQualifiedName(), degreeCurricularPlan.getName(),
-                        executionCourse.getCode());
-                return;
-            }
-
-            String alternativeSiteLink =
-                    row.getCell(COLUMN_ALTERNATIVE_SITE_LINK, Row.CREATE_NULL_AS_BLANK).getStringCellValue().trim();
-
-            if (StringUtils.isEmpty(alternativeSiteLink)) {
-                alternativeSiteLink = null;
-            }
-
-            executionCourse.getSite().setAlternativeSite(alternativeSiteLink);
-        } else {
-            result.addErrorMessage(
-                    "label.org.fenixedu.ulisboa.applications.management.cms.site.alternativesite.process.error.invalidLine",
-                    String.valueOf(row.getRowNum() + 1));
+    private void processFile(File file, ExecutionInterval executionInterval, ProcessResult result) {
+        try {
+            XLSxUtil.importExcel(file, new ExecutionIntervalSheetProcessor(result, executionInterval));
+        } catch (Throwable t) {
+            result.reportFailure();
         }
     }
 
-    private Degree getDegree(Row row, ProcessResult result) {
+    private static Degree getDegree(Row row, ProcessResult result) {
```

Comment:
@joaofreitas-qubit let's move all the methods into the ExecutionIntervalSheetProcessor class that way they don't need to be static.